### ISSUE TITLE
[JENKINS-25912] Workspace remains on disk after job has been ( deleted, moved, renamed )

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-06-a2#5169]
<p>Scenario_1:</p>

<p>Job foo runs once and creates a workspace named "foo".<br/>
After renaming foo to foofoo the workspace "foo" still exists if it wasn't deleted manually.<br/>
If foofoo runs now it creates it's own workspace.</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java"># ls 
foo 
foofoo
</pre>
</div></div>

<p>The problem by having multiple jobs with huge workspaces is the disk space. In addition to that it's not nice to have some dead folders lying around on your file-system.</p>

<p>Scenario_2:<br/>
If job foo is executed in parallel it creates multiple workspaces.<br/>
This is fine as long as you're not using the "Wipe Out Current Workspace" button. <br/>
If so the only workspace that is getting deleted is foo but not the additional ones.</p>

<p>Example <b>before</b> wiping the workspace:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#ls
foo
foo@2
foo@3
foo@4
</pre>
</div></div>

<p>Example <b>after</b> wiping the workspace:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">#ls

foo@2
foo@3
foo@4
</pre>
</div></div>

---
<details><summary><i>Originally reported by <a title="Jan Henne" href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jthenne">jthenne</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-25912" target="_blank">Workspace remains on disk after job has been ( deleted, moved, renamed )</a></i></summary>
<i><ul>
<li><b>reported by</b>: jthenne
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 2
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-11-06
</ul></i>
</details>

**Comment by jenkins-infra-bot** [comment]
<i>[Originally duplicated by: <a href="https://github.com/timja-org/jenkins-gh-issues-poc-11-06-a2/issues?q=in%3Atitle%20JENKINS-25967">JENKINS-25967</a>]</i>
